### DISCUSSION TITLE
Polygon Collision Detection

### DISCUSSION THREAD

**Discussion by tahsinmostafa** [eleurent/highway-env#336] [General]
Hi,

I wanted to know if the `are_polygons_intersecting()` function under `utils.py` guaranteed to be accurate atleast for the `intersecting` boolean variable or there can be some approximation errors? More generally, is the `_is_colliding()` function guaranteed to be accurate to determine if there is a collision or not between an ego vehicle and a surrounding vehicle at a particular frame?

**Comment by eleurent** [comment]
Yes, this function is guaranteed to be correct for the intersection test at the current timestep.
(while the `will_intersect` variable assumes that both polygons will keep moving at constant velocity).

Source and explanations: https://www.codeproject.com/Articles/15573/2D-Polygon-Collision-Detection
